Mathlib/NumberTheory/EllipticDivisibilitySequence.lean

```diff
@@ -64,41 +65,40 @@ elliptic, divisibility, sequence
 
 universe u v w
 
-variable {R : Type u} [CommRing R]
+variable {R : Type u} [CommRing R] (W : ℤ → R)
 
 /-- The proposition that a sequence indexed by integers is an elliptic sequence. -/
-def IsEllSequence (W : ℤ → R) : Prop :=
+def IsEllSequence : Prop :=
   ∀ m n r : ℤ, W (m + n) * W (m - n) * W r ^ 2 =
     W (m + r) * W (m - r) * W n ^ 2 - W (n + r) * W (n - r) * W m ^ 2
 
 /-- The proposition that a sequence indexed by integers is a divisibility sequence. -/
-def IsDivSequence (W : ℤ → R) : Prop :=
+def IsDivSequence : Prop :=
   ∀ m n : ℕ, m ∣ n → W m ∣ W n
 
 /-- The proposition that a sequence indexed by integers is an EDS. -/
-def IsEllDivSequence (W : ℤ → R) : Prop :=
+def IsEllDivSequence : Prop :=
   IsEllSequence W ∧ IsDivSequence W
 
-lemma IsEllSequence_id : IsEllSequence id :=
+lemma isEllSequence_id : IsEllSequence id :=
   fun _ _ _ => by simp only [id_eq]; ring1
 
-lemma IsDivSequence_id : IsDivSequence id :=
+lemma isDivSequence_id : IsDivSequence id :=
   fun _ _ => Int.ofNat_dvd.mpr
 
 /-- The identity sequence is an EDS. -/
-theorem IsEllDivSequence_id : IsEllDivSequence id :=
-  ⟨IsEllSequence_id, IsDivSequence_id⟩
+theorem isEllDivSequence_id : IsEllDivSequence id :=
+  ⟨isEllSequence_id, isDivSequence_id⟩
 
-lemma IsEllSequence_mul (x : R) {W : ℤ → R} (h : IsEllSequence W) : IsEllSequence (x • W) :=
+lemma IsEllSequence.smul (h : IsEllSequence W) (x : R) : IsEllSequence <| x • W :=
   fun m n r => by
     linear_combination (norm := (simp only [Pi.smul_apply, smul_eq_mul]; ring1)) x ^ 4 * h m n r
 
-lemma IsDivSequence_mul (x : R) {W : ℤ → R} (h : IsDivSequence W) : IsDivSequence (x • W) :=
+lemma IsDivSequence.smul (h : IsDivSequence W) (x : R) : IsDivSequence <| x • W :=
   fun m n r => mul_dvd_mul_left x <| h m n r
 
-lemma IsEllDivSequence_mul (x : R) {W : ℤ → R} (h : IsEllDivSequence W) :
-    IsEllDivSequence (x • W) :=
-  ⟨IsEllSequence_mul x h.left, IsDivSequence_mul x h.right⟩
+lemma isEllDivSequence_mul (h : IsEllDivSequence W) (x : R) : IsEllDivSequence <| x • W :=
+  ⟨h.left.smul W x, h.right.smul W x⟩
```

Comment:
Why do you change parentheses to `<|`? It's one character longer and to me it doesn't look as nice. I'd use `<|` only to avoid nested parentheses.
```suggestion
lemma IsEllDivSequence.smul (h : IsEllDivSequence W) (x : R) : IsEllDivSequence (x • W) :=
  ⟨h.left.smul W x, h.right.smul W x⟩
```

Reply:
It's a habit of mine, and I do think mine looks nicer :)

Reply:
Can you elaborate? In paper mathematics, function application is usually written `f(x)` with parentheses, so I'd argue mine is closer to the common mathematics notation. When there are nested parentheses people may use different kinds of delimiters like {}, [] at different levels in addition to (), which are not available in Lean, so `<|` is a nice workaround in this regard.